core/src/main/java/io/grpc/internal/RetryingNameResolver.java

```diff
@@ -95,12 +95,26 @@ public void onResult(ResolutionResult resolutionResult) {
             "RetryingNameResolver can only be used once to wrap a NameResolver");
       }
 
+      // To have retry behavior for name resolvers that haven't migrated to onResult2.
       delegateListener.onResult(resolutionResult.toBuilder().setAttributes(
               resolutionResult.getAttributes().toBuilder()
                   .set(RESOLUTION_RESULT_LISTENER_KEY, new ResolutionResultListener()).build())
           .build());
     }
 
+    @Override
+    public Status onResult2(ResolutionResult resolutionResult) {
+      Status status = delegateListener.onResult2(resolutionResult);
+      if (status != null) { // null indicates LB had been shut down.
+        if (status == Status.OK) {
```

Comment:
There are multiple ways to create `OK` statuses, and they won't all have the same reference equality. You need to use `status.isOk()`.